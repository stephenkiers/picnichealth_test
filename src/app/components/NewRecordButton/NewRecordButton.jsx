import React, {Component} from 'react'
import PropTypes from 'prop-types'


class NewRecordButton extends Component {
    render() {
        return (
            <a
                href="javascript:void(0);"
                className="btn mob-block mob-text-center"
                onClick={this.props.onClick}
            >
                New
            </a>
        )
    }
}
NewRecordButton.defaultProps = {
};
NewRecordButton.propTypes = {
    onClick: PropTypes.func.isRequired,
}

export default NewRecordButton