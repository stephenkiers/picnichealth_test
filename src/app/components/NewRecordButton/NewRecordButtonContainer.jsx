import React, {Component} from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import NewRecordButton from "./NewRecordButton";
import {reduxAppStateSetModalId} from "../../AppState/actions";
import {generateRandomUUID} from "../../utils";


class NewRecordButtonContainer extends Component {
    constructor(props) {
        super(props)

        this.onClick = e => {
            e.preventDefault();
            this.props.setModalId(generateRandomUUID())
        }
    }
    render() {
        return (
            <NewRecordButton
                onClick={this.onClick}
            />
        )
    }
}
NewRecordButtonContainer.defaultProps = {
};
NewRecordButtonContainer.propTypes = {
}


const mapStateToProps = (state, ownProps) => ({
})

const mapDispatchToProps = (dispatch, ownProps) => ({
    setModalId(id) {
        dispatch(reduxAppStateSetModalId(id))
    }
});



export default connect(mapStateToProps, mapDispatchToProps)(NewRecordButtonContainer)