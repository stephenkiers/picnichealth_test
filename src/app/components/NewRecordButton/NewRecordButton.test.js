import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import NewRecordButton from './NewRecordButton';

test('NewRecordButton displays correct', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <NewRecordButton
            onClick={() => {}}
        />
    );
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot();
});