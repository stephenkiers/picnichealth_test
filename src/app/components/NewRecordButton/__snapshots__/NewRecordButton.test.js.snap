// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewRecordButton displays correct 1`] = `
<a
  className="btn mob-block mob-text-center"
  href="javascript:void(0);"
  onClick={[Function]}
>
  New
</a>
`;
